elephant/cell_assembly_detection.py

```diff
@@ -0,0 +1,1143 @@
+        (all assemblies, even if significant, with fewer occurrences
+        than min_occurrences are discarded).
+        Default : 0.
+    size_chunks : int
+        size (in bins) of chunks in which the spike trains is divided
+        to compute the variance (to reduce non stationarity effects
+        on variance estimation)
+        Default : 100.
+    max_spikes : int
+        maximal assembly order (the algorithm will return assemblies of
+        composed by maximum max_spikes elements).
+        Default : numpy.inf
+    significance_pruning : bool
+        if True the method performs significance pruning among
+        the detected assemblies
+        Default: True
+    subgroup_pruning : bool
+        if True the method performs subgroup pruning among
+        the detected assemblies
+        Default: True
+    same_config_cut: bool
+        if True performs pruning (not present in the original code and more
+        efficient), not testing assemblies already formed
+        if they appear in the very same configuration
+        Default: False
+    bool_times_format: bool
+        if True the activation time series is a list of 0/1 elements, where
+        1 indicates the first spike of the patter
+        Otherwise, the activation times of the assemblies are indicated by the
+        indices of the bins in which the first spike of the pattern
+        is happening
+        Default: False
+    verbose: bool
+        Regulates the number of prints given by the method. If true all prints
+        are given, otherwise the method does give any prints.
+        Default: False
+
+    Returns
+    -------
+    assembly_bin : list
+        contains the assemblies detected for the binsize chosen
+        each assembly is a dictionary with attributes:
+        'neurons' : vector of units taking part to the assembly
+                    (unit order correspond to the agglomeration order)
+        'lag' : vector of time lags lag[z] is the activation delay between
+                neurons[1] and neurons[z+1]
+        'pvalue' : vector of pvalues. pvalue[z] is the p-value of the
+                   statistical test between performed adding
+                   neurons[z+1] to the neurons[1:z]
+        'times' : assembly activation time. It reports how many times the
+                  complete assembly activates in that bin.
+                  time always refers to the activation of the first listed
+                  assembly element (neurons[1]), that doesn't necessarily
+                  corresponds to the first unit firing.
+                  The format is  identified by the variable bool_times_format.
+        'signature' : array of two entries (z,c). The first is the number of
+                      neurons participating in the assembly (size),
+                      the second is number of assembly occurrences.
+
+    Raises
+    ------
+    TypeError
+        if the data is not a list of elephant.conv.BinnedSpikeTrains
+    ValueError
```

Comment:
This list is very explicit, I suggest to summarize by saying that `ValueError` is raised if input parameters are out of bounds.